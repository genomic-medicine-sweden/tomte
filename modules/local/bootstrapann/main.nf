process BOOTSTRAPANN {
    tag "$meta.id"
    label 'process_low'

    conda "bioconda::pydamage=0.70"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/pydamage:0.70--pyhdfd78af_0' :
        'biocontainers/pydamage:0.70--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(vcf), path(tbi), path(csv)

    output:
    tuple val(meta), path("*.vcf"), emit: vcf
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    $baseDir/bin/BootstrapAnn.py \\
        --vcf ${vcf} \\
        --ase ${csv} \\
        > ${prefix}_ase.vcf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | sed 's/Python //g')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_ase.vcf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | sed 's/Python //g')
    END_VERSIONS
    """
}
