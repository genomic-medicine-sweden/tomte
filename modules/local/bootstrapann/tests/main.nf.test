nextflow_process {

    name "Test Process BOOTSTRAPANN"
    script "../main.nf"
    process "BOOTSTRAPANN"
    tag "modules_local"
    tag "modules"
    tag "bootstrapann"

    setup {

        run("GATK4_ASEREADCOUNTER") {
            script "../../../nf-core/gatk4/asereadcounter/main.nf"
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path +'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path +'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true),
                    file(params.modules_testdata_base_path +'genomics/sarscov2/illumina/vcf/test.vcf.gz', checkIfExists: true),
                    file(params.modules_testdata_base_path +'genomics/sarscov2/illumina/vcf/test.vcf.gz.tbi', checkIfExists: true)
                ]
                input[1] = [
                    [ id:'reference' ], // meta map
                    file(params.modules_testdata_base_path +'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                input[2] = [
                    [ id:'reference' ], // meta map
                    file(params.modules_testdata_base_path +'genomics/sarscov2/genome/genome.fasta.fai', checkIfExists: true)
                ]
                input[3] = [
                    [ id:'reference' ], // meta map
                    file(params.modules_testdata_base_path +'genomics/sarscov2/genome/genome.dict', checkIfExists: true)
                ]
                input[4] = []
                """
            }
        }
    }

    test("Should run without failures") {

        when {
            params {
                modules_testdata_base_path = 'https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/'
            }
            process {
                """
                input[0] = Channel.of([
                    [ id:'test' ],
                    file(params.modules_testdata_base_path +'genomics/sarscov2/illumina/vcf/test.vcf.gz', checkIfExists: true),
                    file(params.modules_testdata_base_path +'genomics/sarscov2/illumina/vcf/test.vcf.gz.tbi', checkIfExists: true)
                ]).join(GATK4_ASEREADCOUNTER.out.csv)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
