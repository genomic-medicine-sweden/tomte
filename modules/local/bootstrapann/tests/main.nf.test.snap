{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_ase.vcf:md5,e05755ba98284f12bf1df0e6a30e0653"
                    ]
                ],
                "1": [
                    "versions.yml:md5,2b69ef5b6182fde60cf48b73f3ffeec1"
                ],
                "vcf": [
                    [
                        {
                            "id": "test"
                        },
                        "test_ase.vcf:md5,e05755ba98284f12bf1df0e6a30e0653"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2b69ef5b6182fde60cf48b73f3ffeec1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-15T16:15:43.272934"
    }
}